baseline/config/run_training.py

```diff
@@ -0,0 +1,174 @@
+
+    return params
+
+# runs ecole and returns primal-dual integral as a reward
+def runEcole(settings, instance):
+
+    print("New Ecole run with instance ", instance)
+
+    # start a new episode
+    env.reset(instance)
+
+    # get the next action from the given settings
+    action = {k: settings[k] for k in settings}
+
+    for k in action.keys():
+        if action[k] == 'TRUE':
+            action[k] = True
+        elif action[k] == 'FALSE':
+            action[k] = False
+
+    # apply the action and collect the reward
+    _, _, reward, _, _ = env.step(action)
+
+    return reward
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'problem',
+        help='Problem benchmark to process.',
+        choices=['item_placement', 'load_balancing', 'anonymous'],
+    )
+    parser.add_argument(
+        '-t', '--timelimit',
+        help='Instance time limit (in seconds).',
+        default=300,
+        type=float,
+    )
+    parser.add_argument(
+        '-i', '--ninstances',
+        help='Number of training instances.',
+        default=10,
+        type=int,
+    )
+    parser.add_argument(
+        '-e', '--nevaluations',
+        help='Number of function evaluations of SMAC.',
+        default=10,
+        type=int,
+    )
+    args = parser.parse_args()
+
+    # collect the instance files
+    if args.problem == 'item_placement':
+        instances_path = pathlib.Path(f"../../instances/1_item_placement/train/")
+    elif args.problem == 'load_balancing':
+        instances_path = pathlib.Path(f"../../instances/2_load_balancing/train/")
+    elif args.problem == 'anonymous':
+        instances_path = pathlib.Path(f"../../instances/3_anonymous/train/")
+
+    print(f"Processing instances from {instances_path.resolve()}")
+    instance_files = list(instances_path.glob('*.mps.gz'))
+
+    # randomly choose set of training instances
+    instances = random.sample(instance_files, 10)
+
+    # write instances to a file
+    instancefile = open('instances.txt', "a+")
+    for i in range(len(instances)):
+        instancefile.write(str(instances[i]) + '\n')
+    instancefile.close()
+    
+    # create ecole environment
+    env = ec.environment.Configuring(
+          
+          # set time limit for each instance
+          scip_params = {'limits/time' : args.timelimit},
+          
+          # pure bandit, no observation
+          observation_function = None,
+
+          # minimize the primal-dual integral
+          reward_function = ec.reward.PrimalDualIntegral(),
+
+          # collect additional metrics for information purposes
+          information_function = {}
+    )
+
+    # Build Configuration Space which defines all parameters and their ranges
+    cs = ConfigurationSpace()
+    params = getParamsFromFile('parameters.pcs')
+    cs.add_hyperparameters(params)
+
+    # randomly choose set of training instances
+    instances = random.sample(instance_files, args.ninstances)
```

Comment:
Ideally this should be made deterministic, with `random.seed(X)`

Reply:
I made it deterministic.